src/utils/createIconArray.js

```diff
@@ -0,0 +1,14 @@
+import { faStar, faStarHalf } from "@fortawesome/free-solid-svg-icons";
+
+const createIconArray = (rating) => {
+  const iconArray = [];
+  for (let i = 0; i < Math.floor(rating); i++) {
+    iconArray.push(faStar);
+  }
+  if (rating % 1 !== 0) {
+    iconArray.push(faStarHalf);
+  }
+  return iconArray;
```

Comment:
I'm suprised you need a for loop plus an if statement.

Reply:
We are going to merge and work on smaller feature branches, so will look at improving this on a new branch